File: BleModule/src/main/java/com/mpdc4gsr/ble/Scanner.java
package com.mpdc4gsr.ble;

import android.content.Context;

import com.mpdc4gsr.ble.callback.ScanListener;



interface Scanner {
    
    void addScanListener(ScanListener listener);

    void removeScanListener(ScanListener listener);

    void startScan(Context context);

    void stopScan(boolean quietly);

    boolean isScanning();

    void onBluetoothOff();

    void release();

    ScannerType getType();
}

File: BleModule/src/main/java/com/mpdc4gsr/ble/Scanner.kt
package com.mpdc4gsr.ble

import android.content.Context
import com.mpdc4gsr.ble.callback.ScanListener

internal interface Scanner {
    fun addScanListener(listener: ScanListener?)

    fun removeScanListener(listener: ScanListener?)

    fun startScan(context: Context?)

    fun stopScan(quietly: Boolean)

    val isScanning: Boolean

    fun onBluetoothOff()

    fun release()

    val type: ScannerType?
}
